From: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Subject: shmem: convert shmem_fault() to use shmem_get_folio_gfp()
Date: Fri, 2 Sep 2022 20:46:18 +0100

No particular advantage for this function, but necessary to remove
shmem_getpage_gfp().

Link: https://lkml.kernel.org/r/20220902194653.1739778-23-willy@infradead.org
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/shmem.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/mm/shmem.c~shmem-convert-shmem_fault-to-use-shmem_get_folio_gfp
+++ a/mm/shmem.c
@@ -2065,6 +2065,7 @@ static vm_fault_t shmem_fault(struct vm_
 	struct vm_area_struct *vma = vmf->vma;
 	struct inode *inode = file_inode(vma->vm_file);
 	gfp_t gfp = mapping_gfp_mask(inode->i_mapping);
+	struct folio *folio;
 	int err;
 	vm_fault_t ret = VM_FAULT_LOCKED;
 
@@ -2127,10 +2128,11 @@ static vm_fault_t shmem_fault(struct vm_
 		spin_unlock(&inode->i_lock);
 	}
 
-	err = shmem_getpage_gfp(inode, vmf->pgoff, &vmf->page, SGP_CACHE,
+	err = shmem_get_folio_gfp(inode, vmf->pgoff, &folio, SGP_CACHE,
 				  gfp, vma, vmf, &ret);
 	if (err)
 		return vmf_error(err);
+	vmf->page = folio_file_page(folio, vmf->pgoff);
 	return ret;
 }
 
_
